### PR TITLE
Fetch example tests

### PR DESCRIPTION
**Pull request by lastmjs**

### CODE REVIEW

File: examples/outgoing_http_requests/src/index.ts [hunk 1]
```diff
@@ -74,3 +53,52 @@ export default Canister({
         };
     })
 });
+
+async function getXkcdResponse() {
```
Comment (lastmjs) [inline examples/outgoing_http_requests/src/index.ts:57]: Put this back

File: examples/outgoing_http_requests/src/index.ts [hunk 2]
```diff
@@ -74,3 +53,52 @@ export default Canister({
...
+    if (process.env.AZLE_TEST_FETCH) {
+        await fetch(`icp://aaaaa-aa/http_request`, {
```
Comment (lastmjs) [inline examples/outgoing_http_requests/src/index.ts:59]: This is not returning anything

File: examples/outgoing_http_requests/src/index.ts
Comment (lastmjs) [inline examples/outgoing_http_requests/src/index.ts:1]: Change all if statements to elses

File: examples/outgoing_http_requests/src/index.ts [hunk 3]
```diff
@@ -16,7 +16,59 @@ import {
...
+                            get: null
+                        },
+                        headers: [],
+                        body: None,
+                        transform: Some({
+                            function: [ic.id(), 'xkcdTransform'] as [
+                                Principal,
+                                string
+                            ],
+                            context: Uint8Array.from([])
```
[the remark this answers is not in the document]
Comment (bdemann) [reply examples/outgoing_http_requests/src/index.ts:64]: Nope, it's a blob

File: examples/outgoing_http_requests/src/index.ts [hunk 4]
```diff
@@ -54,51 +106,4 @@ export default Canister({
...
-                transform: Some({
-                    function: [ic.id(), 'xkcdTransform'] as [Principal, string],
-                    context: Uint8Array.from([])
-                })
-            }
-        ],
-        cycles: 50_000_000n
-    });
-}
+async function getXkcdResponse() {}
```
[the remark this answers is not in the document]
Comment (bdemann) [reply examples/outgoing_http_requests/src/index.ts:109]: oops

File: examples/outgoing_http_requests/src/index.ts [hunk 5]
```diff
@@ -16,7 +16,59 @@ import {
...
+                                string
+                            ],
+                            context: Uint8Array.from([])
+                        })
+                    }
+                ],
+                cycles: 50_000_000n
+            });
+        }
+    }),
```
[the remark this answers is not in the document]
Comment (bdemann) [reply examples/outgoing_http_requests/src/index.ts:71]: Seems like it

**Pull request merged**